packages/graph-explorer/src/core/StateProvider/localDb.ts

```diff
@@ -0,0 +1,223 @@
+import { z } from "zod";
+import { logger } from "../../utils";
+import { deserializeData, serializeData } from "./serializeData";
+import { fromFileToJson, toJsonFileData } from "../../utils/fileData";
+import saveAs from "file-saver";
+
+/*
+
+# Dev Notes
+
+The backup & restore process is a bit complex. Here is how the process breaks
+down.
+
+## Backup Process
+
+This process will create a JSON file with all the contents of the localForage
```

Comment:
Are we adding a version so that if there are incompatible changes - we can handle them?

Reply:
Yes. It just serves as a marker that can be interpreted during parsing to disambiguate between future iterations.

Those future iterations could be breaking or non-breaking.